WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/WooPaymentSetupWebViewModel.swift

```diff
@@ -0,0 +1,54 @@
+import Foundation
+import WebKit
+
+/// A view model for authenticated web view for setting up WCPay
+final class WooPaymentSetupWebViewModel: AuthenticatedWebViewModel, WebviewReloadable {
+    /// Set in `AuthenticatedWebViewController`.
+    var reloadWebview: () -> Void = {}
+
+    let title: String
+    let initialURL: URL?
+    let completionHandler: (_ isSuccess: Bool) -> Void
+
+    init(title: String = "",
+         initialURL: URL,
+         onCompletion: @escaping (Bool) -> Void) {
+        self.title = title
+        self.initialURL = initialURL
+        self.completionHandler = onCompletion
+    }
+
+    func handleDismissal() {
+        // no-op
+    }
+
+    func handleRedirect(for url: URL?) {
+        guard let urlString = url?.absoluteString else {
+            return
+        }
+        // If the self-hosted site allows login with WPCOM as a Jetpack feature,
+        // WPCOM authentication is complete after redirecting to the WPCOM homepage.
+        if urlString.removingSuffix("/") == Constants.urlAfterWPComAuth,
+           initialURL?.absoluteString != Constants.urlAfterWPComAuth {
+            reloadWebview()
+        }
```

Comment:
Nit: We could consider moving this logic inside `AuthenticatedWebViewModel` as a helper function. 

Something like below,
```swift
extension AuthenticatedWebViewModel {
    func shouldReload(for url: URL?) -> Bool {
        // If the self-hosted site allows login with WPCOM as a Jetpack feature,
        // WPCOM authentication is complete after redirecting to the WPCOM homepage.
        let urlAfterWPComAuth = "https://wordpress.com"
        guard let urlString = url?.absoluteString else {
            return false
        }

        if urlString.removingSuffix("/") == urlAfterWPComAuth,
           initialURL?.absoluteString != urlAfterWPComAuth {
            return true
        }
        return false
    }
}
```

Reply:
Good idea! Added the extension method in ef867d45d3418030974703cde7631085e1be1074 and updated the web view models accordingly.